Libraries/ReactNative/AppRegistry.js

```diff
@@ -237,6 +251,19 @@ const AppRegistry = {
         NativeModules.HeadlessJsTaskSupport.notifyTaskFinished(taskId);
       });
   },
+
+  /**
+   * Only called from native code. Cancels a headless task.
+   *
+   * See http://facebook.github.io/react-native/docs/appregistry.html#cancelheadlesstask
+   */
+  cancelHeadlessTask(taskId: number, taskKey: string): void {
+    const taskCancelProvider = taskCancelProviders.get(taskKey);
+    if (!taskCancelProvider) {
+      throw new Error(`No task canceller registered for key '${taskKey}'`);
+    }
+    taskCancelProvider()();
```

Comment:
>taskCancelProvider()(); [](start = 4, length = 23)

Why did you decide to have two layers of indirection instead of just one. In other words, with the current design people have to do this:

```
function taskCancelProvider() {
  return () => {
    // cancel the task
  };
}
registerCancellableHeadlessTask(someKey, taskProvider, taskCancelProvider);
```

Why not instead design the API like this?:

```
function taskCancelProvider() {
  // cancel the task
}
registerCancellableHeadlessTask(someKey, taskProvider, taskCancelProvider);
```

Reply:
I followed the pattern used for TaskProvider.
The motivation for the original choice seems to be to allow delay loading of the corresponding JS code by having it in a separate JS file and set taskProvider as () => require('SomeTaskName')

See https://facebook.github.io/react-native/docs/headless-js-android.html